test/server/cards/01_SOR/leaders/BobaFettCollectingTheBounty.spec.js

```diff
@@ -0,0 +1,108 @@
+describe('Boba Fett, Collecting the Bounty', function() {
+    integration(function() {
+        describe('Boba Fett\'s leader ability', function() {
+            beforeEach(function() {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['rivals-fall', 'cantina-bouncer'],
+                        groundArena: ['death-star-stormtrooper'],
+                        leader: 'boba-fett#collecting-the-bounty',
+                        base: 'dagobah-swamp',
+                        resources: 6,
+                    },
+                    player2: {
+                        groundArena: ['viper-probe-droid', 'cell-block-guard', 'wampa'],
+                        leader: { card: 'luke-skywalker#faithful-friend', deployed: true },
+                    },
+                });
+            });
+
+            it('should ready a resource when an enemy unit leaves play', function() {
+                const reset = () => {
+                    this.player2.passAction();
+                    this.player1.setResourceCount(6);
+                    this.bobaFett.exhausted = false;
+                };
+
+                // Case 1 - when defeating an enemy unit
+                this.player1.clickCard(this.rivalsFall);
+                this.player1.clickCard(this.viperProbeDroid);
+
+                expect(this.player1.countSpendableResources()).toBe(1);
+                expect(this.bobaFett.exhausted).toBeTrue();
+
+                // Case 2 - when returning an enemy unit to hand
+                reset();
+
+                this.player1.clickCard(this.cantinaBouncer);
```

Comment:
Thanks for adding cantina bouncer for this test, gives me a lot more confidence that everything is working right under the hood